src/os_common/os_fhandle.c

```diff
@@ -355,6 +370,117 @@ __wt_close(WT_SESSION_IMPL *session, WT_FH **fhp)
 	return (__handle_close(session, fh, true));
 }
 
+/*
+ * __wt_fsync_background_chk --
+ *	Return if background fsync is supported.
+ */
+int
+__wt_fsync_background_chk(WT_SESSION_IMPL *session)
+{
+	WT_CONNECTION_IMPL *conn;
+	WT_DECL_RET;
+	WT_FH *fh, *fhnext;
+	WT_FILE_HANDLE *handle;
+
+	conn = S2C(session);
+	WT_ASSERT(session, !F_ISSET(conn, WT_CONN_READONLY));
```

Comment:
With respect to an earlier comment, this might need `WT_CONN_IN_MEMORY`, too. (Or, this is safe, you could just remove the assert as far as I'm concerned.)

Reply:
Removed. It is checked at a higher level.